internal/github/fake_github.go

```diff
@@ -82,44 +85,45 @@ func NewFakeGitHub(h func(output io.Writer, command Command, workingDir string,
 }
 
 func NewAlwaysSucceedsFakeGitHub() *FakeGitHub {
-	return NewFakeGitHub(func(output io.Writer, command Command, workingDir string, fullRepoName string) (bool, error) {
+	return NewFakeGitHub(func(command Command, args []string) (bool, error) {
 		return true, nil
-	}, func(output io.Writer, workingDir string) (interface{}, error) {
+	}, func(workingDir string) (interface{}, error) {
 		return PrStatus{}, nil
 	})
 }
 
 func NewAlwaysFailsFakeGitHub() *FakeGitHub {
-	return NewFakeGitHub(func(output io.Writer, command Command, workingDir string, fullRepoName string) (bool, error) {
+	return NewFakeGitHub(func(command Command, args []string) (bool, error) {
 		return false, errors.New("synthetic error")
-	}, func(output io.Writer, workingDir string) (interface{}, error) {
+	}, func(workingDir string) (interface{}, error) {
 		return nil, errors.New("synthetic error")
 	})
 }
 
 func NewAlwaysThrowNoPRFound() *FakeGitHub {
-	return NewFakeGitHub(func(output io.Writer, command Command, workingDir string, branchName string) (bool, error) {
+	return NewFakeGitHub(func(command Command, args []string) (bool, error) {
+		workingDir, branchName := args[0], args[1]
```

Comment:
How do you know that `args[0]` will always contain `workingDir` and `args[1]` `branchName`?

Reply:
I mean, in this case `NewAlwaysThrowNoPRFound()` happens to just be used in `TestNoPRFound()`, which in turn calls `fakeGitHub`'s `ClosePullRequest()`.

But what if `NewAlwaysThrowNoPRFound()` would happen to call any of the other `fakeGitHub`'s functions that have no branch name in them, such as `Clone()`, which has full repo name in the arguments?

I see the code works as it is now, but we are restricting the reusability of `NewAlwaysThrowNoPRFound()`.

Reply:
It's the same as the original implementation, in which `workingDir `and `branchName` are always expected as arguments. 
At the moment there is only [one usage](https://github.com/Skyscanner/turbolift/blob/main/cmd/updateprs/updateprs_test.go#L55) of this implementation of FakeGitHub, which always calls ClosePullRequest which has those two arguments.
As far as I understand, this implementation isn't expected to be used with any other of the github methods, so it shouldn't be an issue.